File: scripts/browser/run-static.py
"""Retain bounded native-browser frontend checks without cookies or credentials."""
import hashlib,json,pathlib,subprocess,sys
name=sys.argv[2]
assert name in ['startup','transport']
source=(pathlib.Path(__file__).parent/('static-'+name+'-checks.js')).read_text()
r=subprocess.run(['bash',str(pathlib.Path.home()/'.codex/skills/playwright/scripts/playwright_cli.sh'),'-s='+sys.argv[1],'run-code',source],capture_output=True,text=True)
if '### Result\n' not in r.stdout:
 print(r.stdout);print(r.stderr);sys.exit(1)
report=json.loads(r.stdout.split('### Result\n',1)[1].split('\n### ',1)[0]);report['checkSourceSha256']=hashlib.sha256(source.encode()).hexdigest()
body=json.dumps(report,indent=2)+'\n';out=pathlib.Path('artifacts/mvp')/('static-'+name+'-'+hashlib.sha256(body.encode()).hexdigest()+'.json')
with out.open('x') as f:f.write(body)
out.chmod(0o444)
print(json.dumps({'artifact':str(out),'passed':sum(x['pass'] for x in report['results']),'total':len(report['results']),'error':report.get('error')}))

File: scripts/browser/run-static.mjs
// Standalone fallback for batches that exceed the native CLI session lifetime.
import { readFile, writeFile, chmod } from 'node:fs/promises';
import { createHash } from 'node:crypto';
import { pathToFileURL } from 'node:url';
const [runtime,group]=process.argv.slice(2);
if(!runtime||!['startup','transport','handoff','rollout'].includes(group))throw new Error('Provide an installed Playwright module path and startup, transport, handoff or rollout.');
const {chromium}=await import(pathToFileURL(runtime).href),browser=await chromium.launch({channel:'chrome',headless:true});
try{
 const context=await browser.newContext(),page=await context.newPage();
 const source=await readFile(new URL('static-'+group+'-checks.js',import.meta.url),'utf8');
 const report=await (new Function('return ('+source+')'))()(page);
 report.checkSourceSha256=createHash('sha256').update(source).digest('hex');
 const body=JSON.stringify(report,null,2)+'\n',file='artifacts/mvp/static-'+group+'-'+createHash('sha256').update(body).digest('hex')+'.json';
 await writeFile(file,body,{flag:'wx',mode:0o444});await chmod(file,0o444);
 console.log(JSON.stringify({artifact:file,passed:report.results.filter(r=>r.pass).length,total:report.results.length,error:report.error}));
 if(report.error||report.results.some(r=>!r.pass))process.exitCode=1;
}finally{await browser.close();}
